myhomework3.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+require_once 'HttpStatusCodes.php';
+
+//function readHttpLikeInput()
+//{
+//    $f = fopen('php://stdin', 'r');
+//    $store = "";
+//    $toread = 0;
+//
+//    while ($line = fgets($f)) {
+//        $store .= preg_replace("/\r/", "", $line);
+//
+//        if (preg_match('/Content-Length: (\d+)/', $line, $m))
+//            $toread = $m[1] * 1;
+//
+//        if ($line === PHP_EOL)
+//
+//            break;
+//    }
+//
+//    if ($toread > 0)
+//        $store .= fread($f, $toread);
+//
+//    return $store;
+//}
+
+//$contents = readHttpLikeInput();
+
+function outputHttpResponse($statuscode, $statusmessage, $headers, $body)
+{
+    $response = "HTTP/1.1 $statuscode $statusmessage" . PHP_EOL;
+    $response .= "Date: " . date("l, d F  Y h:i:sa") . PHP_EOL;
+    $response .= "Server: Apache/2.2.14 (Win32)" . PHP_EOL;
+    $response .= "Content-Length: " . strlen($body) . PHP_EOL;
+    $response .= "Connection: Closed" . PHP_EOL;
+    $response .= "Content-Type: text/html; charset=utf-8" . PHP_EOL;
+    $response .= PHP_EOL . "$body";
+
+    echo $response;
+}
+
+function processHttpRequest($method, $uri, $headers, $body)
+{
+    try {
+        outputHttpResponse(getStatusCode($method, $uri), "OK", $headers, getResult($uri));
+    } catch (Exception $ex) {
+        outputHttpResponse($ex->getCode(), $ex->getMessage(), $headers, $ex->getMessage());
+    }
+}
+
+function getResult($uri)
+{
+    $sumQueryAndNumbers = explode('=', $uri);
+    $nums = explode(',', $sumQueryAndNumbers[1]);
+
+    return array_sum($nums);
+}
+
+function getStatusCode($method, $uri)
+{
+
+    if (explode("?", $uri)[0] != "/sum") {
```

Comment:
it would be better to validate the request in `processHttpRequest`, cause it's unexpected behaviour for `getStatusCode` function